### DISCUSSION TITLE
How to launch a session on the current open window

### DISCUSSION THREAD

**Discussion by reisnobre** [kovidgoyal/kitty#5327] [Q&A]
Hi, I'm trying to write a couple session configurations for different projects.
I would like to use sessions to be able to jump between each one of those quickly but I didn't found a way to launch the session on top of the current window or overriding the current one.
When the command is called it opens on a new os window. 

Is there a way to open on the current one or attach to the current window?

**Comment by page-down** [comment]
As far as I know `startup_session` does exactly what its name implies, configures the default session for a new instance.
Currently it is not possible to start any group of windows through it.
https://sw.kovidgoyal.net/kitty/conf/#opt-kitty.startup_session

There is no defined concept of "session" in kitty.
If you want to quickly open some kitty windows, run shells or some programs, adjusting the layout, you can do so in several ways.

- Use the combine action if there are not many things to open.
- Use the launch action to start a window that allows remote control and run a script. Launch the kitty windows, tabs you need with the remote control commands. This window will be closed automatically after the script is executed.
- Remote control via socket.
- Write custom kitten and do whatever you want.

> ... on top of the current window or overriding the current one.
> ... open on the current one or attach to the current ...

It sounds like all you need is to close the current OS window, without the confirmation, then open a new one.

**Reply by reisnobre** [reply]
Why then it's possible to create multiple session files and use those to launch kitty as `kitty --session`?.

In what situation should I use this commando but to launch a specific session?

**Reply by page-down** [reply]
> Why then it's possible to ... launch kitty as ...

That will start multiple instances of kitty ("... it opens on a new os window ...") so it won't do what you're asking ("... overriding the current one ...").

Obviously launching new OS windows is not what you want.

> ... what situation should I use ...

When you need to achieve what you asked for in the main post.

https://sw.kovidgoyal.net/kitty/launch/
https://sw.kovidgoyal.net/kitty/actions/#action-combine
https://sw.kovidgoyal.net/kitty/remote-control/#kitty-launch
https://sw.kovidgoyal.net/kitty/kittens/custom/

**Comment by dmatos2012** [comment]
Anyone managed to get this going? I particularly tried with using remote-control but I guess I am really not that experienced, and every time I closed the `remote-control` window of course everything else would close. 

Is it possible then to send commands to the remote-control window itself to close?

**Reply by vanarok** [reply]
I have zsh function. 
```
function php-vite() {
  export PROJECT_DIR=$1
  nohup kitty --session ~/.config/kitty/sessions/php-vite-mailhog.conf & disown
  exit
}
```

**Reply by silverhairs** [reply]
> I have zsh function.
> 
> ```
> function php-vite() {
>   export PROJECT_DIR=$1
>   nohup kitty --session ~/.config/kitty/sessions/php-vite-mailhog.conf & disown
>   exit
> }
> ```

I think this is the best work around so far.
I still think it would have been great to be able to override/extend the active session by merging with the session we are trying to open.
